tests/rptest/clients/rpk.py

```diff
@@ -306,17 +307,27 @@ def parse_field(field_name, string):
             return m['value']
 
         partition_pattern = re.compile(
-            "(?P<topic>.+) +(?P<partition>\d+) +(?P<offset>\d+) +(?P<log_end>\d+) +(?P<lag>\d+) *(?P<member_id>.*)? *(?P<client_id>.*)? *(?P<host>.*)?"
+            "(?P<topic>.+) +(?P<partition>\d+) +(?P<offset>\d+|-) +(?P<log_end>\d+|-) +(?P<lag>-?\d+|-) *(?P<member_id>.*)? *(?P<client_id>.*)? *(?P<host>.*)?"
         )
 
         def parse_partition(string):
             m = partition_pattern.match(string)
             assert m is not None, f"Partition string '{string}' does not match the pattern"
+
+            # Account for negative numbers and '-' value
+            all_digits = lambda x: x.lstrip('-').isdigit()
+
+            offset = int(m['offset']) if all_digits(
+                m['offset']) else m['offset']
+            log_end = int(m['log_end']) if all_digits(
+                m['log_end']) else m['log_end']
+            lag = int(m['lag']) if all_digits(m['lag']) else m['lag']
```

Comment:
Group describe can return '-' character which means that no information was fetched for that partition.

These changes along with the regex updates on line 310 were necessary to account for the '-' character.